jetty-core/jetty-http2/jetty-http2-tests/src/test/java/org/eclipse/jetty/http2/tests/HTTP2Test.java

```diff
@@ -1139,6 +1140,113 @@ public void onClose(Session session, GoAwayFrame frame, Callback callback)
         assertThat(failure.getMessage(), containsString("invalid_hpack_block"));
     }
 
+    @Test
+    public void testClientCreatesStreamsWithExplicitStreamId() throws Exception
+    {
+        start(new ServerSessionListener() {});
+
+        Session session = newClientSession(new Session.Listener() {});
+
+        int evenStreamId = 128;
+        MetaData.Request request = newRequest("GET", HttpFields.EMPTY);
+        assertThrows(ExecutionException.class, () -> session.newStream(new HeadersFrame(evenStreamId, request, null, true), new Stream.Listener() {})
+            .get(5, TimeUnit.SECONDS));
+
+        // Equivalent to Integer.MAX_VALUE + 2.
+        int negativeStreamId = Integer.MIN_VALUE + 1;
```

Comment:
Why not using `Integer.MAX_VALUE + 2` then?

Reply:
Because it is not immediately clear that it would overflow and become negative.